elasticsearch/connection/http_urllib3.py

```diff
@@ -111,6 +111,8 @@ def perform_request(self, method, url, params=None, body=None, timeout=None, ign
             if not isinstance(method, str):
                 method = method.encode('utf-8')
 
+            request_headers = dict(self.headers)
+            request_headers.update(headers or {})
```

Comment:
It would be nice if this block only ran `if headers:` to avoid copying the `self.headers` dict every time when it is only needed for the bulk-like bodies